app/models/katello/sync_plan.rb

```diff
@@ -85,15 +90,13 @@ def next_sync_date
           end
           next_sync = now.advance(:minutes => minutes, :seconds => seconds)
         when DAILY
-          sync_time = Time.at(self.sync_date.hour * 60 * 60 + self.sync_date.min * 60 + self.sync_date.sec)
-          now_time = Time.at(now.hour * 60 * 60 + now.min * 60 + now.sec)
-          if sync_time < now_time
+          if self.next_sync_time < now_time
             hours += 24
           end
           next_sync = now.advance(:hours => hours, :minutes => minutes, :seconds => seconds)
         when WEEKLY
           days = self.sync_date.wday - now.wday
-          days += 7 if days <= 0
+          days += 7 if days <= 0 && self.next_sync_time <= now_time
```

Comment:
Shouldn't it be more like:

`if days < 0 || (days == 0 &&  self.next_sync_time <= now_time)  `

we wouldn't want to ignore this if days is -1, but the sync time is less than the current time.